pyqtgraph/functions.py

```diff
@@ -947,73 +951,48 @@ def rescaleData(data, scale, offset, dtype=None, clip=None):
-        try:
-            import scipy.weave
-        except ImportError:
-            raise Exception('scipy.weave is not importable; falling back to slower version.')
-        
-        ## require native dtype when using weave
-        if not data.dtype.isnative:
-            data = data.astype(data.dtype.newbyteorder('='))
-        if not dtype.isnative:
-            weaveDtype = dtype.newbyteorder('=')
+    # p = np.poly1d([scale, -offset*scale])
+    # d2 = p(data)
+    d2 = data.astype(np.float) - float(offset)
+    d2 *= scale
+
+    # Clip before converting dtype to avoid overflow
+    if dtype.kind in 'ui':
+        lim = np.iinfo(dtype)
+        if clip is None:
+            # don't let rescale cause integer overflow
+            d2 = np.clip(d2, lim.min, lim.max)
         else:
-            weaveDtype = dtype
-        
-        newData = np.empty((data.size,), dtype=weaveDtype)
-        flat = np.ascontiguousarray(data).reshape(data.size)
-        size = data.size
-        
-        code = """
-        double sc = (double)scale;
-        double off = (double)offset;
-        for( int i=0; i<size; i++ ) {
-            newData[i] = ((double)flat[i] - off) * sc;
-        }
-        """
-        scipy.weave.inline(code, ['flat', 'newData', 'size', 'offset', 'scale'], compiler='gcc')
-        if dtype != weaveDtype:
-            newData = newData.astype(dtype)
-        data = newData.reshape(data.shape)
-    except:
-        if getConfigOption('useWeave'):
-            if getConfigOption('weaveDebug'):
-                debug.printExc("Error; disabling weave.")
-            setConfigOptions(useWeave=False)
-        
-        #p = np.poly1d([scale, -offset*scale])
-        #d2 = p(data)
-        d2 = data - float(offset)
-        d2 *= scale
-        
-        # Clip before converting dtype to avoid overflow
-        if dtype.kind in 'ui':
-            lim = np.iinfo(dtype)
-            if clip is None:
-                # don't let rescale cause integer overflow
-                d2 = np.clip(d2, lim.min, lim.max)
-            else:
-                d2 = np.clip(d2, max(clip[0], lim.min), min(clip[1], lim.max))
-        else:
-            if clip is not None:
-                d2 = np.clip(d2, *clip)
-        data = d2.astype(dtype)
+            d2 = np.clip(d2, max(clip[0], lim.min), min(clip[1], lim.max))
+    else:
+        if clip is not None:
+            d2 = np.clip(d2, *clip)
+    data = d2.astype(dtype)
     return data
-    
+
+
 def applyLookupTable(data, lut):
     """
     Uses values in *data* as indexes to select values from *lut*.
     The returned data has shape data.shape + lut.shape[1:]
     
     Note: color gradient lookup tables can be generated using GradientWidget.
+
+    Parameters
+    ----------
+    lut : ndarray
+        Either cupy or numpy arrays are accepted, though this function has
+        problems in cupy on windows with some versions of the cuda toolkit.
```

Comment:
is this with cuda versions < 11.0? ...probably should try and be a bit more specific here (doesn't have to be exhaustive, but documenting what's been observed would be good).